app/assets/javascripts/discourse/app/components/post/menu/buttons/reply.gjs

```diff
@@ -1,23 +1,107 @@
 import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
 import { service } from "@ember/service";
 import DButton from "discourse/components/d-button";
 import concatClass from "discourse/helpers/concat-class";
+import {
+  invalidateDraftState,
+  matchesPost,
+  setDraftFromTopic,
+  setDraftSaved,
+} from "discourse/lib/draft-state-cache";
 import { i18n } from "discourse-i18n";
 
+/**
+ * @component PostMenuReplyButton
+ *
+ * Renders the per-post reply action button. When a topic-level reply draft exists,
+ * the button shows "Open Draft" and a corresponding tooltip; otherwise it shows "Reply".
+ *
+ * @param {object} post - The post this button belongs to (this.args.post)
+ * @param {object} state - Menu state (this.args.state)
+ * @param {object} buttonActions - Reply action handlers
+ */
 export default class PostMenuReplyButton extends Component {
   static shouldRender(args) {
     return args.state.canCreatePost;
   }
 
   @service site;
+  @service appEvents;
 
+  @tracked hasMatchingDraft = false;
+
+  constructor() {
```

Comment:
I think if you move more stuff into a service you can get rid of all this finagling in this component